### PR TITLE
Brought rrig's implementation of kerberos delegation up to date with master

### PR DESCRIPTION
**Pull request by JoshuaWarburton**
Made the changes as suggested here a8a6ada1c143e5b339480b3d509b456a619ffae3 and merged with master

### CODE REVIEW

**Comment by JoshuaWarburton** [comment]
Is there anything more that needs to be done to merge this, or are we just waiting on approval?

File: requests_kerberos/kerberos_.py [hunk 1]
```diff
@@ -96,11 +99,17 @@ def generate_request_header(self, response):
...
+        # Flags used by kerberos module.
+        gssflags = kerberos.GSS_C_MUTUAL_FLAG | kerberos.GSS_C_SEQUENCE_FLAG
+        if self.delegate:
+            gssflags |= kerberos.GSS_C_DELEG_FLAG
+
         try:
             result, self.context[host] = kerberos.authGSSClientInit(
-                "{0}@{1}".format(self.service, host))
+                "{0}@{1}".format(self.service, host), gssflags=gssflags)
         except kerberos.GSSError:
```
Comment (kevints) [inline requests_kerberos/kerberos_.py]: `except kerberos.GSSError as e:`?

Comment (JoshuaWarburton) [inline requests_kerberos/kerberos_.py]: Is that strictly necessary as we don't do anything with "e"?

Comment (kevints) [inline requests_kerberos/kerberos_.py]: You attempt to log it below

Comment (JoshuaWarburton) [inline requests_kerberos/kerberos_.py]: Ahh, yes. I was reading one except block too far down. Considering we log the exception here, it should probably be logged in the other blocks too.

**Comment by JoshuaWarburton** [comment]
New commit to address comments from @kevints

**Comment by mkomitee** [comment]
I've verified that the tests pass and that delegation works.

It also merges cleanly.

I'd really like to see this extended to use GSS_C_DELEG_POLICY_FLAG, but it's not available in kerberos (or pykerberos), so I don't think we should require it.

This looks good to me.

:+1:

**Comment by sigmavirus24** [comment]
Thanks for the review @mkomitee

**Pull request merged**